extensions/wikia/BannerNotifications/js/BannerNotifications.js

```diff
@@ -0,0 +1,269 @@
+		types = {
+			'notify': 'blue',
+			'confirm': 'green',
+			'error': 'red',
+			'warn': 'yellow'
+		},
+		classes = Object.keys(types).join(' '),
+		pageContainer,
+		wikiaHeader,
+		headerHeight,
+		modal,
+		isModal;
+
+	/**
+	 * Constructs jQuery element with the notification
+	 * @param {String} content
+	 * @param {String} [type]
+	 * @returns {jQuery}
+	 */
+	function createMarkup(content, type) {
+		return $('<div class="banner-notification">' +
+			'<button class="close wikia-chiclet-button">' +
+			'<img src="' + window.stylepath + '/oasis/images/icon_close.png">' +
+			'</button><div class="msg">' + content + '</div></div>')
+			.addClass(type)
+			.hide();
+	}
+
+	/**
+	 * Adds given markup to DOM
+	 * @param {jQuery} $element
+	 * @param {jQuery} $parentElement
+	 */
+	function addToDOM($element, $parentElement) {
+		// allow notification wrapper element to be passed by extension
+		if ($parentElement instanceof jQuery) {
+			$parentElement.prepend($element);
+
+			// handle modal implementations
+		} else if (isModal) {
+			modal.prepend($element);
+
+			// handle non-modal implementation
+		} else {
+			$(pageContainer).prepend($element);
+		}
+
+		$element.fadeIn('slow');
+	}
+
+	/**
+	 * Main entry point for this feature - shows the notification
+	 * and returns the notification instance
+	 * @param {string} content - message to be displayed
+	 * @param {string} type - See BannerNotifications.options for supported types
+	 * @param {jQuery} [$parent] Element to prepend notification to
+	 * @param {number} [timeout] Optional time (in ms) after which notification will disappear.
+	 */
+
+	function show(content, type, $parent, timeout) {
+		var bannerNotification;
+
+		/**
+		 * Removes notification with a fade-out animation
+		 * @param {Function} callback
+		 */
+		function hide(callback) {
+			removeFromDOM(bannerNotification.$element, callback);
+		}
+
+		/**
+		 * Changes content of the notification to the provided one
+		 * @param {String} content
+		 */
+		function setContent(content) {
+			bannerNotification
+				.$element
+				.find('.msg')
+				.html(content);
+		}
+
+		/**
+		 * Changes type of the notification to the provided one
+		 * @param {String} type
+		 */
+		function setType(type) {
+			if (types.hasOwnProperty(type)) {
+				bannerNotification
+					.$element
+					.removeClass(classes)
+					.addClass(type);
+			}
+		}
+
+		bannerNotification = {
+			$element: createMarkup(content, type),
+			hide: hide,
+			setContent: setContent,
+			setType: setType
+		};
+
+		isModal = isModalShown();
+
+		// Modal notifications have no close button so set a timeout
+		if (isModal && typeof timeout !== 'number') {
+			timeout = defaultTimeout;
+		}
+
+		bannerNotification.setType(type);
+
+		addToDOM(
+			bannerNotification.$element,
+			$parent
+		);
+
+		// Share scroll event with WikiaFooterApp's toolbar floating (BugId:33365)
+		if (window.WikiaFooterApp) {
+			window.WikiaFooterApp.addScrollEvent();
```

Comment:
Did you look into removing this dependency? I can see why they did it this way for performance reasons, but it might be nice to abstract the scroll event handling into a separate module that both `WikiaFooterApp` and `BannerNotifications` could plug into. The current dependency seems brittle and unexpected from the point of view of someone looking at this for the first time.